pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -324,18 +344,19 @@ func (e *cloudWatchExecutor) QueryData(ctx context.Context, req *backend.QueryDa
 		frontend, but because alerts are executed on the backend the logic needs to be reimplemented here.
 	*/
 	q := req.Queries[0]
-	model, err := simplejson.NewJson(q.JSON)
+	var model DataQueryJson
+	err := json.Unmarshal(q.JSON, &model)
 	if err != nil {
 		return nil, err
 	}
 	_, fromAlert := req.Headers["FromAlert"]
-	isLogAlertQuery := fromAlert && model.Get("queryMode").MustString("") == "Logs"
+	isLogAlertQuery := fromAlert && model.QueryMode == "Logs"
 
 	if isLogAlertQuery {
 		return e.executeLogAlertQuery(ctx, req)
 	}
 
-	queryType := model.Get("type").MustString("")
+	queryType := model.QueryType
 
 	var result *backend.QueryDataResponse
 	switch queryType {
```

Comment:
Can model.QueryType just get plugged in here?

Reply:
Yep, since `QueryType` will just return `""` if it's not set. Which is what `MustString` is doing.